File: Modul6/Bahasa_Python/PRAK601-2310817210012-Nazmi-Hakim.py
def buatmatriks(baris, kolom, isi):
    matriks = []
    index = 0

    for i in range(baris):
        barismatriks = []
        for j in range(kolom):
            barismatriks.append(int(isi[index]))
            index += 1
        matriks.append(barismatriks)

    return matriks

def cetakmatriks(matriks):
    for baris in matriks:
        for elemen in baris:
            print(elemen, end=' ')
        print()

baris, kolom = map(int, input().split())
elemenmatriks = input().split()
matriks = buatmatriks(baris, kolom, elemenmatriks)
cetakmatriks(matriks)

File: Modul6/Bahasa_C/PRAK601-2310817210012-Nazmi-Hakim.c
#include <stdio.h>

void buat_matriks(int baris, int kolom, int isi[], int matriks[baris][kolom]) {
    int index = 0;

    for (int i = 0; i < baris; i++) {
        for (int j = 0; j < kolom; j++) {
            matriks[i][j] = isi[index++];
        }
    }
}

void cetak_matriks(int baris, int kolom, int matriks[baris][kolom]) {
    for (int i = 0; i < baris; i++) {
        for (int j = 0; j < kolom; j++) {
            printf("%d ", matriks[i][j]);
        }
        printf("\n");
    }
}

int main() {
    int baris, kolom;
    scanf("%d %d", &baris, &kolom);

    int elemenmatriks[baris * kolom];
    for (int i = 0; i < baris * kolom; i++) {
        scanf("%d", &elemenmatriks[i]);
    }

    int matriks[baris][kolom];
    buat_matriks(baris, kolom, elemenmatriks, matriks);

    cetak_matriks(baris, kolom, matriks);
    return 0;
}